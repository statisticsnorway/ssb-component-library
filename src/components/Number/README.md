# Number

> A component for showing number in different size
> The number should be properly formatted

### Usage

#### HTML

```html
<div class="ssb-number small">123</div>

<div class="ssb-number medium">123 456</div>

<div class="ssb-number large">123 456 789</div>
```

#### React

```jsx harmony
<Number size="small">123</Number>

<Number size="medium">123 456</Number>

<Number size="large">123 456 789</Number>
```

Available props:

| Name      | Type                         | Description              |
| --------- | ---------------------------- | ------------------------ |
| children  | node                         | Renders a node           |
| className | string                       | Optional container class |
| size      | 'small', 'medium' or 'large' | Change size on number    |
