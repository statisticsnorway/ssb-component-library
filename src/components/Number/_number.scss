$large-number-size: 150px;
$medium-number-size: 100px;
$small-number-size: 50px;

.ssb-number {
  @include roboto-condenced;
  color: $ssb-dark-5;
  line-height: 1;

  &.small {
    font-size: $small-number-size;
    line-height: 1.1;
  }

  &.medium {
    font-size: $medium-number-size;
  }

  &.large {
    font-size: $large-number-size;
  }
}