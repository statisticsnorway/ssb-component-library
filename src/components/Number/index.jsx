import React from 'react';
import PropTypes from 'prop-types';

const Number = ({
	children, className, size,
}) => (
	<div className={`ssb-number ${size}${className ? ` ${className}` : ''}`}>
		{children}
	</div>
);

Number.defaultProps = {
	size: 'small',
};

Number.propTypes = {
	children: PropTypes.node,
	className: PropTypes.string,
	size: PropTypes.oneOf(['small', 'medium', 'large']),
};

export default Number;
