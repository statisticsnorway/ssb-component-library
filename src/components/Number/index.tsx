import React, { type ReactNode } from 'react'
import classNames from '../../utils/utils'

interface NumberProps {
  children?: ReactNode
  className?: string
  size?: 'small' | 'medium' | 'large'
}

const Number: React.FC<NumberProps> = ({ children, className, size = 'small' }) => (
  <div className={classNames('ssb-number', size, className)}>{children}</div>
)

export default Number
