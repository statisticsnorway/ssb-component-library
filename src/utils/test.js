import fs from 'fs';
import path from 'path';
import { render as rtlRender } from '@testing-library/react';

export * from '@testing-library/react';
export * from '@testing-library/jest-dom';

const styles = fs.readFileSync(path.relative(process.cwd(), './lib/bundle.css'), 'utf8');

export function render(ui, options) {
	const view = rtlRender(ui, options);

	const styleElement = document.createElement('style');
	styleElement.innerHTML = styles;
	document.body.appendChild(styleElement);

	return view;
}
