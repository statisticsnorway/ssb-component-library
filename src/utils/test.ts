import fs from 'fs'
import path from 'path'
import { RenderOptions, RenderResult, render as rtlRender } from '@testing-library/react'
import { ReactNode } from 'react'

export * from '@testing-library/react'

const styles = fs.readFileSync(path.relative(process.cwd(), './lib/bundle.css'), 'utf8')

export function render(ui: ReactNode, options?: RenderOptions): RenderResult {
  const view = rtlRender(ui, options)

  const styleElement = document.createElement('style')
  styleElement.innerHTML = styles
  document.body.appendChild(styleElement)

  return view
}
