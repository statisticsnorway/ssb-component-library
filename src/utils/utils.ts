// utils.ts
const classNames = (...classes: (string | boolean | undefined)[]): string => classes.filter(Boolean).join(' ')

export default classNames
