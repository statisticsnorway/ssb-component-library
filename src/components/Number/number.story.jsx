import React from 'react'
import Number from './index'

export default {
  title: 'Number',
}

export const Small = () => (
  <div>
    <Number size='small'>599 246 391</Number>
  </div>
)

export const Medium = () => (
  <div>
    <Number size='medium'>59 923</Number>
  </div>
)

export const Large = () => (
  <div>
    <Number size='large'>789 398</Number>
  </div>
)
