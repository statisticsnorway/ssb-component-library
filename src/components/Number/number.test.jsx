import React from 'react'

import { render } from '../../utils/test'
import Number from './index'

describe('Number component', () => {
  test('Matches the snapshot', () => {
    const { asFragment } = render(<Number>599 246 391</Number>)
    expect(asFragment()).toMatchSnapshot()
  })
  test('Toggles classNames correctly', () => {
    const { asFragment } = render(<Number size='small'>599 246 391</Number>)
    expect(asFragment()).toMatchSnapshot()
  })
})
