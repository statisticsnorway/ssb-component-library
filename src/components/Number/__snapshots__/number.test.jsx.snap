// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Number component Matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="ssb-number small"
  >
    599 246 391
  </div>
</DocumentFragment>
`;

exports[`Number component Toggles classNames correctly 1`] = `
<DocumentFragment>
  <div
    class="ssb-number small"
  >
    599 246 391
  </div>
</DocumentFragment>
`;
